### ISSUE TITLE
 BET/examples/linearMap/linearMapUniformSampling.py

### ISSUE THREAD

**Issue by mpilosov** [UT-CHG/BET#221]
There appears to be an issue with calling ```sampler.random_sample_set``` on an ```sample_set``` object in that it doesn't write to the sample object that's passed in unless explicitly declared as output. 
This can be seen by running the example in the title of this issue using ```ipython``` and looking at the memory locations of the input_samples variable and the output of the ```random_sample_set (or regular)``` procedure. 
```
In [5]: input_samples.get_values()
```
nothing... weird. but the prior information is still there...
```
In [6]: input_samples.get_dim()
Out[6]: 3

In [7]: input_samples.get_domain()
Out[7]: 
array([[ 0.,  1.],
       [ 0.,  1.],
       [ 0.,  1.]])
In [8]: sampler.regular_sample_set(input_samples, num_samples_per_dim=[15, 15, 10])
Out[8]: <bet.sample.sample_set at 0x7f15100453d0>

In [9]: input_samples
Out[9]: <bet.sample.sample_set at 0x7f14fff27350>
```
ah so they're in different locations. let's try
```
In [10]: input_samples = sampler.regular_sample_set(input_samples, num_samples_per_dim=[15, 15, 10])

In [11]: input_samples
Out[11]: <bet.sample.sample_set at 0x7f1510045590>

In [12]: input_samples.get_values()
Out[12]: 
array([[ 0.0625    ,  0.0625    ,  0.09090909],
       [ 0.0625    ,  0.0625    ,  0.18181818],
       [ 0.0625    ,  0.0625    ,  0.27272727],
       ..., 
       [ 0.9375    ,  0.9375    ,  0.72727273],
       [ 0.9375    ,  0.9375    ,  0.81818182],
       [ 0.9375    ,  0.9375    ,  0.90909091]])

```
Yes now it works!
So in the example, changing lines 63-67 to:
```
randomSampling = True
if randomSampling is True:
    input_samples = sampler.random_sample_set('random', input_samples, num_samples=1E3)
else:
    input_samples = sampler.regular_sample_set(input_samples, num_samples_per_dim=[15, 15, 10])
```

allows the code to run. I know this shouldn't be necessary to set the output like that, but I suspect something in the ```basic.Sampling.random_sample_set``` method (and the ```regular_sample_set``` one too) is not preserving pointers. I think in ```basicSampling``` that lines 84 and again at 143 (perhaps elsewhere too) is the culprit.
```
    # check to see what the input object is
    if isinstance(input_obj, sample.sample_set):
        input_sample_set = input_obj.copy()
```
It makes a copy of the ```input_object``` called ```input_sample_set``` and works on it to set the values and returns them, but it's now called something different than ```input_obj```, so without explicitly defining the output on the LHS like I did above, it's no surprise the error occurred. I assume it got introduced a couple commits ago when the input variables to this method got changed. 

Do note that the same exact issue arises in the nonlinear example as well. Should the output just be declared explicitly and the examples/linear and examples/nonlinear files be changed, or the method in basicSampling?

**Comment by lcgraham** [comment]
We decided to not alter the 'input_obj' in place but rather return a new 'sample_set' object as a design choice. This is so that the user can create multiple sample sets and discretizations using different sampling techniques without overwriting the input. You should update the example and create a pull-request.

**Comment by mpilosov** [comment]
Perfect! That makes a lot of sense. Thank you for the explanation. I will do that. I hope I get the pull request right, bear with me if I don't, please. =)

**Comment by lcgraham** [comment]
This should remain open till #222 is merged.

**Comment by mpilosov** [comment]
So just to clarify, this issue comes up in two files 
```examples/linear/linearMapUniformSampling.py``` and ```examples/nonlinear/nonlinearMapUniformSampling``` so I have made the change in those two places. I wasn't messing around with whitespace in those at all. 
Furthermore I changed the ```myModel.py``` file in examples/linear/ because it had a typo with two variable assignments ```QoI_samples = data = ... ``` in line 9. In that file I also just added one whitespace after a comma in the same line. 

Are the plethora of commits referring to the fact that I added a file in that branch earlier as well? (73552233b33d1a97cb6f1f692031e4b4e1dd895e) or prior ones referring to me deleting compilation files and whatnot (did I not clean up correctly?)
